src/crimson/osd/osd.cc

```diff
@@ -1402,6 +1403,93 @@ seastar::future<> OSD::send_beacon()
   return monc->send_message(std::move(beacon));
 }
 
+seastar::future<> OSD::run_bench(int64_t count, int64_t bsize, int64_t osize, int64_t onum, double* elapsed) {
+  LOG_PREFIX(OSD::run_bench);
+  DEBUG(" count={}, bsize={}, osize={}, onum={}", count, bsize, osize, onum);
+
+  return seastar::do_with(ceph::os::Transaction(), ceph::os::Transaction(),
+    [=, this](auto& t, auto& cleanup) {
+    return seastar::do_for_each(boost::make_counting_iterator<int64_t>(0),
+      boost::make_counting_iterator<int64_t>(onum),
+      [&t, &cleanup, osize](int64_t i) {
+        std::string data(osize, 'a');
+        ceph::buffer::list bl(osize);
+        bl.append(data);
+        std::string oid_str = fmt::format("disk_bw_test_{}", i);
+        ghobject_t oid(hobject_t(sobject_t(object_t(oid_str), 0)),
+          ghobject_t::NO_GEN,
+          shard_id_t::NO_SHARD);
+        t.write(coll_t::meta(), oid, 0, data.size(), bl);
+        cleanup.remove(coll_t::meta(), oid);
+        return seastar::make_ready_future<>();
+      }).then([this, &t, &cleanup]() {
+        return store.get_sharded_store().open_collection(coll_t::meta()).then(
+          [this, &t, &cleanup](auto collection_ref) {
+            return store.get_sharded_store().do_transaction(
+              collection_ref, std::move(t)).then([this, &cleanup, collection_ref] {
+                return store.get_sharded_store().do_transaction(
+                  collection_ref, std::move(cleanup)).then([collection_ref] {
```

Comment:
Cleanup phase should not be part of the test results (elapsed time), we probably shouldn't submit `cleanup` here.